python/ray/data/dataset.py

```diff
@@ -270,14 +270,15 @@ def map(
             >>> import ray
             >>> # Transform python objects.
             >>> ds = ray.data.range(1000)
-            >>> ds.map(lambda x: x * 2)
+            >>> ds.map(lambda x: x * 2).fully_executed()
             Map
```

Comment:
`Map` shouldn't be shown up after `fully_executed()` call.

Reply:
Actually this PR should not impact behavior for `ray.data.range()`. Why we need `fully_executed()` here?